tests/gatewayDerive.ts

```diff
@@ -68,39 +88,182 @@ describe("gateway-derive", () => {
     it("creates a new derived pass", async () => {
       const [, derivedPass] = await service.derivePass(sourceGknKeys);
 
-      const derivedPassAccount = await program.account.derivedPass.fetch(derivedPass);
+      const derivedPassAccount = await program.account.derivedPass.fetch(
+        derivedPass
+      );
       expect(derivedPassAccount.sourceGkns).to.deep.equal(sourceGknKeys);
     });
   });
 
-  context('derived pass issuance', () => {
-    beforeEach('generate the derived pass', async () => {
-      const authorityService = new DerivedPassService(program, authorityProvider);
+  context("derived pass issuance", () => {
+    beforeEach("generate the derived pass", async () => {
+      const authorityService = new DerivedPassService(
+        program,
+        authorityProvider
+      );
       [, derivedPass] = await authorityService.derivePass(sourceGknKeys);
 
       service = new DerivedPassService(program, ownerProvider);
-    })
+    });
 
-    context("with no passes in the owner's wallet", ()=> {
+    context("with no passes in the owner's wallet", () => {
       it("should not be able to derive a pass", async () => {
         const shouldFail = service.issue(authority, derivedPass);
 
         return expect(shouldFail).to.be.rejectedWith(/MissingComponentPass/);
       });
     });
 
-    context("when the owner has the requisite component passes", ()=> {
-      beforeEach('issue the component passes', async () => {
-        await Promise.all(civicGatekeeperServices.map(gks =>
-          sendGatewayTransaction(() => gks.issue(owner.publicKey))
-        ));
+    context("when the owner has the requisite component passes", () => {
+      beforeEach("issue the component passes", async () => {
+        await Promise.all(
+          civicGatekeeperServices.map((gks) =>
+            sendGatewayTransaction(() => gks.issue(owner.publicKey))
+          )
+        );
       });
 
       it("should be able to derive a pass", async () => {
         const [, gatewayToken] = await service.issue(authority, derivedPass);
 
-        const foundToken = await findGatewayToken(authorityProvider.connection, owner.publicKey, derivedPass);
-        expect(foundToken.publicKey.toBase58()).to.equal(gatewayToken.toBase58());
+        const foundToken = await findGatewayToken(
+          authorityProvider.connection,
+          owner.publicKey,
+          derivedPass
+        );
+        expect(foundToken.publicKey.toBase58()).to.equal(
+          gatewayToken.toBase58()
+        );
+      });
+
+      context("with fees", () => {
+        // fees for the two constituent passes
+        const fee0 = 100;
+        const fee1 = 1000;
```

Comment:
I'd really love to see a fee = 0; and fee = -X; test in the Ts layer (while latter should probably be rejected rather than overflow) and fee = 0 works as expected. (even if it does not formally unset).